secant/Wrappers/WrappedSDKSynchronizer.swift

```diff
@@ -0,0 +1,552 @@
+//
+//  WrappedSDKSynchronizer.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 13.04.2022.
+//
+
+import Foundation
+import ZcashLightClientKit
+import Combine
+import ComposableArchitecture
+
+enum WrappedSDKSynchronizerState: Equatable {
+    case unknown
+    case transactionsUpdated
+    case started
+    case progressUpdated
+    case statusWillUpdate
+    case synced
+    case stopped
+    case disconnected
+    case syncing
+    case downloading
+    case validating
+    case scanning
+    case enhancing
+    case fetching
+    case minedTransaction
+    case foundTransactions
+    case failed
+    case connectionStateChanged
+}
+
+enum WrappedSDKSynchronizerError: Error {
+    case synchronizedNotInitialized
+}
+
+protocol WrappedSDKSynchronizer {
+    var blockProcessor: CompactBlockProcessor? { get }
+    var notificationCenter: WrappedNotificationCenter { get }
+    var synchronizer: SDKSynchronizer? { get }
+    var stateChanged: CurrentValueSubject<WrappedSDKSynchronizerState, Never> { get }
+    var walletBirthday: BlockHeight? { get }
+    var latestScannedSynchronizerState: SDKSynchronizer.SynchronizerState? { get }
+
+    func prepareWith(initializer: Initializer, seedBytes: [UInt8]) async throws
+    func start(retry: Bool) throws
+    func stop()
+    func synchronizerSynced(_ synchronizerState: SDKSynchronizer.SynchronizerState?)
+    func statusSnapshot() -> SyncStatusSnapshot
+
+    func rewind(_ policy: RewindPolicy) async throws
+    
+    func getShieldedBalance() -> WalletBalance?
+    func getTransparentBalance() -> WalletBalance?
+    func getAllClearedTransactions() -> Effect<[WalletEvent], Never>
+    func getAllPendingTransactions() -> Effect<[WalletEvent], Never>
+    func getAllTransactions() -> Effect<[WalletEvent], Never>
+
+    func getTransparentAddress(account: Int) async -> TransparentAddress?
+    func getSaplingAddress(account: Int) async -> SaplingAddress?
+    
+    func sendTransaction(
+        with spendingKey: UnifiedSpendingKey,
+        zatoshi: Zatoshi,
+        to recipientAddress: Recipient,
+        memo: String?
+    ) -> Effect<Result<TransactionState, NSError>, Never>
+}
+
+extension WrappedSDKSynchronizer {
+    func start() throws {
+        try start(retry: false)
+    }
+
+    func getTransparentAddress() async -> TransparentAddress? {
+        await getTransparentAddress(account: 0)
+    }
+
+    func getSaplingAddress() async -> SaplingAddress? {
+        await getSaplingAddress(account: 0)
+    }
+}
+
+class LiveWrappedSDKSynchronizer: WrappedSDKSynchronizer {
+    private var cancellables: [AnyCancellable] = []
+    private(set) var blockProcessor: CompactBlockProcessor?
```

Comment:
Why do we need a reference to the processor ?

Reply:
It's gone now.